File: models/Gpt-3.5-turbo/cpp/code/problems/additional_problems/Food_Division.cpp

#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> currentFood(n);
    vector<int> requiredFood(n);

    for (int i = 0; i < n; i++) {
        cin >> currentFood[i];
    }

    for (int i = 0; i < n; i++) {
        cin >> requiredFood[i];
    }

    int minSteps = 0;

    for (int i = 0; i < n; i++) {
        int diff = requiredFood[i] - currentFood[i];
        if (diff > 0) {
            int nextChild = (i + 1) % n;
            currentFood[i] += diff;
            currentFood[nextChild] -= diff;
            minSteps += diff;
        }
    }

    cout << minSteps << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/additional_problems/Food_Division.py

n = int(input())
current_food = list(map(int, input().split()))
required_food = list(map(int, input().split()))

excess_food = 0
for i in range(n):
    diff = current_food[i] - required_food[i]
    if diff > 0:
        excess_food += diff

steps = excess_food // n
print(steps)
